client/main.lua

```diff
@@ -92,90 +108,151 @@ local function GetDeliveryLocation()
+end
+
+local function updateZone()
+    if table.type(PlayerData) == 'empty' or PlayerData.job.name ~= 'bus' then
+        if VehicleZone then
+            VehicleZone:remove()
+            VehicleZone = nil
+        end
+        return
+    end
+
+    if not VehicleZone then
+        local inRange = false
+        local shownTextUI = false
+        local inVeh = whitelistedVehicle()
+        VehicleZone = lib.zones.sphere({
+            name = "qb_busjob_bus_main",
+            coords = Config.Location.xyz,
+            radius = 5,
+            debug = false,
+            onEnter = function()
+                inRange = true
+                CreateThread(function()
+                    repeat
+                        Wait(0)
+                        if not inVeh then
+                            if not shownTextUI then
+                                lib.showTextUI(Lang:t('info.busstop_text'))
+                                shownTextUI = true
+                            end
+                            if IsControlJustReleased(0, 38) then
+                                busGarage()
+                                lib.hideTextUI()
+                                shownTextUI = false
+                                break
+                            end
+                        else
+                            if not shownTextUI then
+                                lib.showTextUI(Lang:t('info.bus_stop_work'))
+                                shownTextUI = true
+                            end
+                            if IsControlJustReleased(0, 38) then
+                                if not NpcData.Active or NpcData.Active and not NpcData.NpcTaken then
+                                    if cache.vehicle then
+                                        BusData.Active = false
+                                        DeleteVehicle(cache.vehicle)
+                                        RemoveBlip(NpcData.NpcBlip)
+                                        lib.hideTextUI()
+                                        shownTextUI = false
+                                        resetNpcTask()
+                                        break
+                                    end
+                                else
+                                    QBCore.Functions.Notify(Lang:t('error.drop_off_passengers'), 'error')
+                                end
+                            end
+                        end
+                    until not inRange
+                end)
+            end,
+            onExit = function()
+                lib.hideTextUI()
+                shownTextUI = false
+                inRange = false
+            end
+        })
+    end
 end
 
 RegisterNetEvent("qb-busjob:client:TakeVehicle", function(data)
-    local coords = Config.Location
-    if (BusData.Active) then
+    if BusData.Active then
         QBCore.Functions.Notify(Lang:t('error.one_bus_active'), 'error')
         return
     else
```

Comment:
I know you didn't edit this, but removing the ```else```  will improve appearance, considering the linked ```if``` statement includes a return.